go-controller/pkg/node/gateway_localnet.go

```diff
@@ -273,12 +283,16 @@ func localnetIptRules(svc *kapi.Service, gatewayIP string) []iptRule {
 type localnetNodePortWatcherData struct {
 	ipt       util.IPTablesHelper
 	gatewayIP string
+	isIPv6    bool
 }
 
 func (npw *localnetNodePortWatcherData) addService(svc *kapi.Service) error {
```

Comment:
I think with the latest fix, IP tables rules are installed based on cluster IP which fixes the core issue. However we are registering two call backs for each service one for v4 and one for v6. One of them is a no-op. Hence for every service created there will be two call backs in case of dual stack , one of which is not needed.  This is functionally not causing any issue, but I will leave it up to you  if this causes any scaling issues. It would be good to optimize if not in this PR but in a future PR.